addons/account_edi_ubl_cii/models/account_edi_common.py

```diff
@@ -373,17 +373,29 @@ def _import_partner(self, company_id, name, phone, email, vat, country_code=Fals
         partner = self.env['res.partner'] \
             .with_company(company_id) \
             ._retrieve_partner(name=name, phone=phone, email=email, vat=vat, domain=domain)
+        country_code = postal_address.get('country_code')
+        country = self.env.ref(f'base.{country_code.lower()}', raise_if_not_found=False) if country_code else self.env['res.country']
+        state_code = postal_address.get('state_code')
+        state = self.env['res.country.state'].search(
+            [('country_id', '=', country.id), ('code', '=', state_code)]
+        ) if state_code and country else self.env['res.country.state']
         if not partner and name and vat:
             partner_vals = {'name': name, 'email': email, 'phone': phone}
             if peppol_eas and peppol_endpoint:
                 partner_vals.update({'peppol_eas': peppol_eas, 'peppol_endpoint': peppol_endpoint})
-            country = self.env.ref(f'base.{country_code.lower()}', raise_if_not_found=False) if country_code else False
-            if country:
-                partner_vals['country_id'] = country.id
             partner = self.env['res.partner'].create(partner_vals)
             if vat and self.env['res.partner']._run_vat_test(vat, country, partner.is_company):
                 partner.vat = vat
             logs.append(_("Could not retrieve a partner corresponding to '%s'. A new partner was created.", name))
+        if not partner.country_id and not partner.street and not partner.street2 and not partner.city and not partner.zip and not partner.state_id:
```

Comment:
Not asking you to do it, but an alternative notation:
`not all([partner.country_id, partner.street, ...])
Some like it, some don't. I personally like it :laughing:

Reply:
IMO, it adds a little confusion 😅